test/e2e/prometheus_test.go

```diff
@@ -189,6 +190,34 @@ func TestPrometheusRemoteWrite(t *testing.T) {
       keySecret:
         name: %[2]s
         key: client.key
+`,
+			expected: []remoteWriteTest{
+				{
+					query:       fmt.Sprintf(`ceil(delta(prometheus_remote_storage_samples_pending{pod="%[1]s",prometheus_replica="%[1]s"}[1m]))`, "prometheus-k8s-0"),
+					expected:    func(v int) bool { return v != 0 },
+					description: "prometheus_remote_storage_samples_pending indicates no remote write progress, expected a continuously changing delta",
+				},
+				{
+					query:       fmt.Sprintf(`ceil(delta(prometheus_remote_storage_samples_pending{pod="%[1]s",prometheus_replica="%[1]s"}[1m]))`, "prometheus-k8s-1"),
+					expected:    func(v int) bool { return v != 0 },
+					description: "prometheus_remote_storage_samples_pending indicates no remote write progress, expected a continuously changing delta",
+				},
+			},
+		},
+		{
+			name: "assert remote write with Authorization works",
+			rwSpec: `
+  - url: https://%[1]s/api/v1/write
+    tlsConfig:
+      ca:
+        secret:
+          name: %[2]s
+          key: ca.crt
+    authorization:
+      type: Bearer
+      credentials:
+        name: %[2]s
+        key: token
```

Comment:
Does this actually test token authorization? I thought we determined I got ahead of myself with suggesting an e2e test for this :thinking:

Reply:
Indeed sorry I forgot to update